File: composeApp/src/androidMain/kotlin/com/exploramus/app/composables/screens/details/DetailsScreenEvents.kt
package com.exploramus.app.composables.screens.details

import com.exploramus.app.composables.navigation.controller.ScreenNavActions
import com.exploramus.shared.viewmodel.core.Events
import com.exploramus.shared.viewmodel.screens.details.singledetail.toggleFavorite

sealed class DetailsUiEvent {
    data class ToggleFavorite(val countryId: String) : DetailsUiEvent()
    object OnBackClicked : DetailsUiEvent()
}

class DetailsEventHandler(
    val events: Events,
    val navActions: ScreenNavActions,
) {
    fun onEvent(event: DetailsUiEvent) {
        when (event) {
            is DetailsUiEvent.ToggleFavorite -> events.toggleFavorite(event.countryId)
            is DetailsUiEvent.OnBackClicked -> navActions.navigateBack()
        }
    }
}

File: iosApp/iosApp/views/screens/details/DetailsScreenEvents.swift

import SwiftUI
import ComposeApp

enum DetailsUiEvent {
    case toggleFavorite(String)
}

struct DetailsEventHandler {
    let events: Events
    
    func onEvent(_ event: DetailsUiEvent) {
        switch event {
        case .toggleFavorite(let id):
            events.toggleFavorite(code: id)
        }
    }
}

